### PR TITLE
ci: Improve stage/job names

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Comment by phlax** [comment]
arggh - kinda unbelievable - the "required" jobs are linked by their label names not the tag (only microsoft)

im gonna have to revert the required job labels to avoid a lot of disruption

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -361,10 +374,11 @@ stages:
         echo "linux_arm64 released"
 
 - stage: check
+  # displayName: Checks (Linux x64)
   dependsOn: ["env", "linux_x64"]
   jobs:
   - job: bazel
-    displayName: "linux_x64"
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:367]: revert

**Comment by adisuissa** [comment]
@phlax overall LGTM, would suggest avoiding the display-names that are commented out.
LMK when you think this is ready for review

**Comment by phlax** [comment]
>  would suggest avoiding the display-names that are commented out.

i kinda wanted to keep them - so we do add them in the future and the stubs are there ready

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!

**Comment by phlax** [comment]
apologies @adisuissa it seems there was one last name to ~revert

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**